packages/appsync-modelgen-plugin/src/visitors/appsync-visitor.ts

```diff
@@ -788,7 +788,16 @@ export class AppSyncModelVisitor<
    * @param field
    */
   protected isRequiredField(field: CodeGenField): boolean | undefined {
-    return !(this.config.handleListNullabilityTransparently ? (field.isList ? field.isListNullable : field.isNullable) : field.isNullable);
+    const hasDefaultDirective = this.hasDefaultDirective(field);
+    return !(this.config.handleListNullabilityTransparently
+      ? field.isList
+        ? field.isListNullable
+        : field.isNullable || hasDefaultDirective
+      : field.isNullable || hasDefaultDirective);
+  }
+
+  protected hasDefaultDirective(field: CodeGenField): boolean {
+    return !!field.directives.find(directive => directive.name === 'default');
```

Comment:
You can use `field.directives.some` rather than find if you just want to get a boolean result back, then we don't need the `!!`